### PR TITLE
Feature/jude/create-staffs-methods-and-routes

### PR DESCRIPTION
**Pull request by judeokafor**
An authenticated and authorized  user as an admin can create staff member, get all staff members, get an individual staff member and also deactivate or activate staff member or user by id

### CODE REVIEW

**Review by codemutatio — APPROVED** [review]
Looks good to me and is ready to be merged

File: functions/src/controllers/staffs.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	 * @param {object} res - response object
+	 * @return  {Object} result
+	 */
+    static async activateOrDeactivateStaff(req, res) {
+        try {
+            const { id } = req.params;
+            const { status, type } = req.query;
+            await db.doc(`/users/${id}`).update({ status: status });
+            return successNoData(res, OK,
+                // eslint-disable-next-line max-len
```
Comment (enyeInc) [inline functions/src/controllers/staffs.js:89]: stop disabling the lint. They are there for a reason

File: functions/src/controllers/staffs.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                `${type === 'staff' ? 'Staff' : 'User'} ${status === 'active' ? 'activated' : 'deactivated'} successfully`);
```
Comment (enyeInc) [inline functions/src/controllers/staffs.js:90]: also this is an anti-pattern. Do not nest your ternaries. Use and `if/else if/else` it is clearer.

File: functions/src/controllers/users.js [hunk 3]
```diff
@@ -108,12 +106,28 @@ class userController {
...
+  * @function
+  * @param {object} req - request object
+  * @param {object} res - response object
+  * @return  {Object} result
+  */
+    static async logoutUser(req, res) {
+        try {
+            const { userId } = req.user;
+            await firebase.auth().signOut();
+            return successNoData(res, 200, 'Logout successfully');
```
Comment (enyeInc) [inline functions/src/controllers/users.js:128]: http-status-code for 200

File: functions/src/controllers/staffs.js (hunk 2 above)
Comment (judeokafor) [reply functions/src/controllers/staffs.js:90]: Alright. Will take care of that

File: functions/src/controllers/users.js (hunk 3 above)
Comment (judeokafor) [reply functions/src/controllers/users.js:128]: Will implement that in my new pr

File: functions/src/controllers/staffs.js (hunk 1 above)
Comment (judeokafor) [reply functions/src/controllers/staffs.js:89]: Alright. I actually forgot to uncomment after i refactored

File: functions/src/controllers/users.js (hunk 3 above)
Comment (enyeInc) [reply functions/src/controllers/users.js:128]: no implement it in this PR.

**Pull request merged**